dae/dae/annotation/annotate_vcf.py

```diff
@@ -27,78 +34,150 @@ def configure_argument_parser() -> argparse.ArgumentParser:
-            print(str(vcf_var), file=out_file, end="")
-
+                vcf_var.info[attribute] = buff
+            out_file.write(vcf_var)
+    out_file.close()
     in_file.close()
 
-    if args.output != "-":
-        out_file.close()
+
+def combine(input_filepath, pipeline, part_filenames, output_filepath):
+    """Combine annotated region parts into a single VCF file."""
+    input_file = VariantFile(input_filepath)
+    update_header(input_file, pipeline)
+    with VariantFile(output_filepath, "w", header=input_file.header) as out:
+        for part_filename in part_filenames:
+            part_file = VariantFile(part_filename)
+            for rec in part_file.fetch():
+                out.write(rec)
+    input_file.close()
+
+
+def produce_regions(context, contigs, region_size):
+    """Given a region size, produce contig regions to annotate by."""
+    ref_genome = context.get_reference_genome()
+    assert ref_genome is not None
+    contig_lengths = {}
+    unknown_contigs = []
+    for contig in contigs:
+        try:
+            contig_lengths[contig] = ref_genome.get_chrom_length(contig)
+        except ValueError:
+            logger.warning(
+                "Could not find contig %s in reference genome", contig
+            )
+            unknown_contigs.append((contig,))
+    return [
+        (contig, start, start + region_size)
+        for contig, length in contig_lengths.items()
+        for start in range(1, length, region_size)
+    ] + unknown_contigs
+
+
+def produce_part_filenames(input_filepath, regions, work_dir):
+    """Produce a list of filenames for output region part files."""
+    filenames = []
+    for region in regions:
+        filenames.append(os.path.join(work_dir, PART_FILENAME.format(
+            in_file=os.path.basename(input_filepath),
+            chrom=region[0], pos_beg=region[1], pos_end=region[2]
+        )))
+    return filenames
+
+
+def cli(raw_args: Optional[list[str]] = None) -> None:
+    """Run command line interface for annotate_vcf tool."""
+    if not raw_args:
+        raw_args = sys.argv[1:]
+
+    parser = configure_argument_parser()
+    args = parser.parse_args(raw_args)
+    VerbosityConfiguration.set(args)
+    CLIAnnotationContext.register(args)
+
+    context = get_genomic_context()
+    pipeline = CLIAnnotationContext.get_pipeline(context)
+
+    if args.output:
+        output = args.output
+    else:
+        output = os.path.basename(args.input).split(".")[0] + "_annotated.vcf"
+
+    if not os.path.exists(args.work_dir):
+        os.mkdir(args.work_dir)
+
+    def run_parallelized():
+        with TabixFile(args.input) as pysam_file:
+            contigs = list(map(str, pysam_file.contigs))
+        regions = produce_regions(context, contigs, args.region_size)
+        filenames = produce_part_filenames(args.input, regions, args.work_dir)
+        task_graph = TaskGraph()
+        for index, (region, filepath) in enumerate(zip(regions, filenames)):
```

Comment:
`filepath` probably should be just `filename`?

Reply:
After thinking about this, I'll actually move to rename `filenames` to `file_paths` (here and in other places where it's relevant) as these are supposed to be relevant or absolute paths pointing to a file, instead of just a file name. This should help clarify the meaning of the code.